kube/base/templates/4xx-rate-deployment.tmpl

```diff
@@ -0,0 +1,37 @@
+---
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  name: {{.Namespace}}-{{.Name}}-4xx-rate-deployment
+  namespace: {{.Namespace}}
+  labels:
+    prometheus: {{.NSPrometheus}}
+    role: alert-rules
+spec:
+  groups:
+  - name: {{.Namespace}}-{{.Name}}-4xx-rate-deployment.rules
+    rules:
+    - alert: {{.Name}}-4xx-rate-deployment
+      annotations:
+        summary: |
+          {{.Identifier}}: 4xx proportion above {{.Threshold}} for 1m
+      expr: |
+        (
+          sum(
+            rate(
+              http_requests_total{namespace="{{.Namespace}}",app="{{.Selector}}",status=~"4.."}[30s]
```

Comment:
We don't want to just do `app="{{.Selector}}"`. We have lots of apps that are deployed with labels like this:

```
app: my-app
env: staging
```

This template would select the metrics for both envs in that case since it doesn't take the other labels into account.

We need to transform the `deployment.Spec.Selector` into a set of label selectors on the metric. For the above we would want to generate:
```
{app="my-app",env="staging"}
```
for example.